eth_abi/decoding.py

```diff
@@ -340,6 +341,74 @@ def from_type_str(cls, abi_type, registry):
         return cls.as_decoder(value_bit_size=abi_type.sub * 8)
 
 
+class BaseFixedDecoder(Fixed32ByteSizeDecoder):
+    frac_places = None
+    is_big_endian = True
+
+    @classmethod
+    def validate(cls):
+        super().validate()
+
+        if cls.frac_places is None:
+            raise ValueError("must specify `frac_places`")
+
+        if not (cls.frac_places > 0 or cls.frac_places <= 80):
```

Comment:
do you find the following easier to parse?

```python
if frac_places <= 0 or frac_places > 80:  # I like this one more because it doesn't have a `not` in it.
    pass
if not 0 < frac_places <= 80:
    pass

Reply:
It seems a bit easier to go straight from the ABI spec to the if statement with not. I didn't even realize the second option you gave was valid syntax. I'm sure Guido hates it but I kinda like it in this case. 😄

I'll switch it to that.

Reply:
@carver sorry I didn't see your comment. So you don't like the `not 0 < frac <= 80` option? The problem I have with the other option is, if you've just come from the ABI spec, it's hard to flip the greater than and less than equal in your mind. Seems like an easy way to get confused. I saw that if clause as checking whether or not an assumption laid out in the spec did not hold.

As for the modular congruence, I think that happens in the super call but I'll double check when I get back home.

Reply:
Actually I just realized that code in the parens is totally wrong. It should be an `and` not an `or`.

Reply:
I don't like `not 0 < frac <= 80` mostly because it's too many operations on one line, plus order of operations is not visibly apparent. If you really wanted to go the route of defining valid, and taking the inverse, I'd prefer:

```py
is_valid_decimal_places = (0 < frac <= 80)
if not is_valid_decimal_places:
    raise
```

These are all things that are easy to keep tweaking over time, though, so I wouldn't get too hung up on it. Except for asking the question of what new test to write to catch:

> Actually I just realized that code in the parens is totally wrong. It should be an and not an or.